fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/JDBCIdentityCacheImpl.java

```diff
@@ -160,6 +164,38 @@ public Long getCommonTokenValueId(String codeSystem, String tokenValue) {
         return result;
     }
 
+    @Override
+    public Set<Long> getCommonTokenValueIds(Collection<CommonTokenValue> tokenValues) {
+        Set<CommonTokenValue> misses = new HashSet<>();
+        Set<Long> result = cache.getResourceReferenceCache().resolveCommonTokenValueIds(tokenValues, misses);
+
+        if (misses.isEmpty()) {
+            return result;
+        }
+
+        if (logger.isLoggable(Level.FINE)) {
+            logger.fine("Cache miss. Fetching common_token_value_ids from database: " + misses);
+        }
+
+        Set<CommonTokenValueResult> readCommonTokenValueIds = resourceReferenceDAO.readCommonTokenValueIds(misses);
+        result.addAll(readCommonTokenValueIds.stream()
```

Comment:
just that it was a tad slower in code with very tight loops.  I don't think we need to avoid ALL uses of streams because of that.

Reply:
this particular method should happen once per search parameter and typically we don't have a lot of those.

Reply:
Yeah, just thought I'm mention it in case you thought it could have an impact. It's a shame about the performance, because the syntactic convenience is nice. Agree we probably don't have to worry unless it shows up in a profile down the road.